src/pyclaw/classic/solver.py

```diff
@@ -126,15 +117,30 @@ def step(self,solution):
         :Output: 
          - (bool) - True if full step succeeded, False otherwise
         """
-        
-        self.cfl.set_global_max(0.)
 
-        if self.before_step is not None:
-            self.before_step(self,solution.states[0])
+        # Choose new time step
```

Comment:
Thanks, that's a good point. At first I thought it wasn't possible, but I figure it out. At the beginning of step(), the function get_dt() that lives in the base solver is called so that avoids duplicating code.